### ISSUE TITLE
[YSQL] DataGrip failed to connect to YB cluster

### ISSUE THREAD

**Issue by d-uspenskiy** [yugabyte/yugabyte-db#3168]
After connect to DB DataGrip fetches schema information with the following query:

```
select N.oid::bigint as id,
       N.xmin as state_number,
       nspname as name,
       D.description,
       pg_catalog.pg_get_userbyid(N.nspowner) as "owner"
from pg_catalog.pg_namespace N
  left join pg_catalog.pg_description D on N.oid = D.objoid
/* */where N.nspname not like 'pg_toast%' and N.nspname not like 'pg_temp%' 
order by case when nspname = current_schema then -1::bigint else N.oid::bigint end
```

This query fails as `xmin` system column is not yet supported.
```ERROR:  System column with id -3 is not supported yet```

**Comment by sstubbs** [comment]
I have the same problem.

**Comment by tsmull-11** [comment]
There is a workaround for this issue which would be to add select the "Introspect using JDBC metadata" box when setting up the DB connection. We need to use this option because YugabyteDB does not use PostgreSQL's system columns given that these functions are handled by the DocDB storage engine of YugabyteDB. I've attached a blog post that goes over how we set up datagrip with YB. Here is a blog that goes over this as well

https://blog.yugabyte.com/getting-started-with-jetbrains-datagrip-on-a-distributed-sql-database/
